src/Model/Report.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Model;
+
+use Friendica\Core\Logger;
+use Friendica\Database\DBA;
+
+class Report
```

Comment:
Can you move this class to `src\Moderation\Repository\Report` and extend it from `BaseRepository`.
So you can make the function like `public function store()` and use `$this->database` instead of all `DBA::` calls.

And then you can pass this `Report` class as an argument at `src\Module\Api\Mastodo\Reports.php`.
There you can copy the behavior of such classes like at the other Mastodon API classes.

If you want, I can make adapt it that way at this PR, but I think this could be a good entrypoint for you at this "new Paradigm" thingy .. but as always just as u like :-)

Reply:
Currently I feel like the last tinker on the Titanic. The mail system has to be improved so that moderators can send private messages to non followers, reports need to be transmitted and received from remote systems, a new user class "moderator" needs to be introduced, there is the need for frontend changes to perform reports, the moderator backend has to be done, ... And in addition to all of these, some nasty errors need to be fixed as well.

In short: I have no to little time at the moment to learn something new.

Reply:
The moderation overhaul is a large endeavor, please get in touch so that we agree on features before we have any major code down to minimize the potential rewrites.

Reply:
Definitely. There is a lot of stuff to think about even on the moderation backend, like some protocol, so that every moderator can see what had been done with each report.

Reply:
Yes, I'd like use to have a shared design document before we go further with code. This is fine because it is about a Mastodon API endpoint, but anything further requires some sort of plan. What is the most convenient way to do this? Collabora? Google Documents? Also would you like to use instant messages to talk about it?

Reply:
Surely, we can use Matrix for that as well. My "plan" (I never have a plan, that's why I'm using quotes here) is to start small but then expand without having legacy issues.

Reply:
I've registered a Matrix account on an Austrian homeserver and I found two of your accounts and none are the one you have in your profile currently.